### DISCUSSION TITLE
skip/run test based on environment

### DISCUSSION THREAD

**Discussion by brc-dd** [vitest-dev/vitest#1959] [Q&A] [answered]
I want to skip/run a test only if the edge-runtime is being used. Is there any way to do that?

**Comment by sheremet-va** [comment, marked as answer]
you can use `describe.runIf` 🤔

**Reply by brc-dd** [reply]
Yeah, but inside that I'd need to write some custom logic to detect the environment. I wanted to know if there is a direct way to get environment passed while running (like `vitest run --environment edge-runtime`) using some context/globals. Thanks anyway!

BTW, `describe.runIf` isn't documented, just `test.runIf` is.

**Reply by sheremet-va** [reply]
I don’t think we expose it publicly, but you can just run command with env: for example, `EDGE=true vitest run --environment edge-runtime`

**Reply by sheremet-va** [reply]
By the way we now expose it publicly through `expect.getState().environment` 😄

**Reply by brc-dd** [reply]
Cool! 🙌
